src/arbcalls/acb_mat.jl

```diff
@@ -0,0 +1,147 @@
+arbcall"void acb_mat_scalar_mul_acb(acb_mat_t B, const acb_mat_t A, const acb_t c, slong prec)"
+arbcall"void acb_mat_scalar_div_si(acb_mat_t B, const acb_mat_t A, slong c, slong prec)"
+# arbcall"void acb_mat_scalar_div_fmpz(acb_mat_t B, const acb_mat_t A, const fmpz_t c, slong prec)"
+arbcall"void acb_mat_scalar_div_arb(acb_mat_t B, const acb_mat_t A, const arb_t c, slong prec)"
+arbcall"void acb_mat_scalar_div_acb(acb_mat_t B, const acb_mat_t A, const acb_t c, slong prec)"
+
+### Gaussian elimination and solving
+arbcall"int acb_mat_lu_classical(slong * perm, acb_mat_t LU, const acb_mat_t A, slong prec)"
+arbcall"int acb_mat_lu_recursive(slong * perm, acb_mat_t LU, const acb_mat_t A, slong prec)"
+arbcall"int acb_mat_lu(slong * perm, acb_mat_t LU, const acb_mat_t A, slong prec)"
+arbcall"void acb_mat_solve_tril_classical(acb_mat_t X, const acb_mat_t L, const acb_mat_t B, int unit, slong prec)"
+arbcall"void acb_mat_solve_tril_recursive(acb_mat_t X, const acb_mat_t L, const acb_mat_t B, int unit, slong prec)"
+arbcall"void acb_mat_solve_tril(acb_mat_t X, const acb_mat_t L, const acb_mat_t B, int unit, slong prec)"
+arbcall"void acb_mat_solve_triu_classical(acb_mat_t X, const acb_mat_t U, const acb_mat_t B, int unit, slong prec)"
+arbcall"void acb_mat_solve_triu_recursive(acb_mat_t X, const acb_mat_t U, const acb_mat_t B, int unit, slong prec)"
+arbcall"void acb_mat_solve_triu(acb_mat_t X, const acb_mat_t U, const acb_mat_t B, int unit, slong prec)"
+arbcall"void acb_mat_solve_lu_precomp(acb_mat_t X, const slong * perm, const acb_mat_t LU, const acb_mat_t B, slong prec)"
+arbcall"int acb_mat_solve(acb_mat_t X, const acb_mat_t A, const acb_mat_t B, slong prec)"
+arbcall"int acb_mat_solve_lu(acb_mat_t X, const acb_mat_t A, const acb_mat_t B, slong prec)"
+arbcall"int acb_mat_solve_precond(acb_mat_t X, const acb_mat_t A, const acb_mat_t B, slong prec)"
+arbcall"int acb_mat_inv(acb_mat_t X, const acb_mat_t A, slong prec)"
+arbcall"void acb_mat_det_lu(acb_t det, const acb_mat_t A, slong prec)"
+arbcall"void acb_mat_det_precond(acb_t det, const acb_mat_t A, slong prec)"
+arbcall"void acb_mat_det(acb_t det, const acb_mat_t A, slong prec)"
+arbcall"void acb_mat_approx_solve_triu(acb_mat_t X, const acb_mat_t U, const acb_mat_t B, int unit, slong prec)"
+arbcall"void acb_mat_approx_solve_tril(acb_mat_t X, const acb_mat_t L, const acb_mat_t B, int unit, slong prec)"
+arbcall"int acb_mat_approx_lu(slong * P, acb_mat_t LU, const acb_mat_t A, slong prec)"
+arbcall"void acb_mat_approx_solve_lu_precomp(acb_mat_t X, const slong * perm, const acb_mat_t A, const acb_mat_t B, slong prec)"
+arbcall"int acb_mat_approx_solve(acb_mat_t X, const acb_mat_t A, const acb_mat_t B, slong prec)"
+arbcall"int acb_mat_approx_inv(acb_mat_t X, const acb_mat_t A, slong prec)"
+
+### Characteristic polynomial and companion matrix
+# arbcall"void _acb_mat_charpoly(acb_ptr poly, const acb_mat_t mat, slong prec)"
```

Comment:
I am not yet fully sure what an `acb_ptr` is and how we should handle this

Reply:
`acb_ptr` (and `arb_ptr`) are the vector types in Arb. I started some work on supporting them but it got a bit messy and I switched to implementing matrices. For now I think we can ignore them and add them later on if we implement support for them. `acb_srcptr` is equivalent to `const acb_ptr` so from Julias perspective they are the same.

Reply:
I can take my `AcbVector` prototype and update it for use in here. Should I commit here, or to a separate branch @saschatimme ?

Reply:
However, you prefer.